src/main/java/com/pnudev/librarysystem/entity/Borrowing.java

```diff
@@ -30,12 +33,16 @@ public class Borrowing {
     private Long id;
 
     @ManyToOne
-    @JoinColumn(name = "user_id")
+    @JoinColumn(name = "user_id", updatable = false, insertable = false)
     private User user;
+    @Column(name = "user_id")
+    private Long userId;
```

Comment:
I am not sure whether you need both user and userId

Reply:
I used it to create a Borrowing (in the reserveBook method of the BorrowingService) in order not to make unnecessary calls to db and UserService. There is another way to do it (as far as I know): create an empty User and set an id (e.g, 
`Borrowing.builder()`
`.user(User.builder().id(userId).build())`
....). 
Is it better? Maybe you have some suggestions?

Reply:
I was basically thinking about something like a https://vladmihalcea.com/the-best-way-to-map-a-onetomany-association-with-jpa-and-hibernate/ for bidirectional one-to-many
Saving both `user` and `userId` is over-redundant. Using only `userId` will require service/repository calls every time you will require a user so I think just going with `user` should suffice.

Reply:
Is it good now?